src/app/_components/countdown/countdown.tsx

```diff
@@ -0,0 +1,54 @@
+"use client";
+import { useEffect, useState } from "react";
+import CountdownClock from "@/app/_components/countdown/countdown-clock.tsx";
+import CountdownComplete from "./countdown-complete";
+
+type CountdownProps = {
+  eventStartTime: Date;
+  timeRemaining: number;
+};
+export default function Countdown({
+  eventStartTime,
+  timeRemaining,
+}: CountdownProps) {
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [time, setTime] = useState(timeRemaining);
+
+  useEffect(() => {
+    const diff = eventStartTime.valueOf() - Date.now();
+    if (!isLoaded) {
+      setIsLoaded(true);
+      if (diff > 0) setTime(diff);
+      else return;
+    }
+    const timerID = setTimeout(() => {
+      setTime(diff);
+    }, 1000);
+
+    if (time < -43200000) {
```

Comment:
This constant value should be referenced under a constant variable. or the expression itself should be assigned a constant variable.

```ts
const 12_Hour_MS = 43200000
const event_has_completed = time < -12_Hour_MS 
const event_still_active = time > -12_Hour_MS && time < 0)
```

Reply:
Gotcha. Done so in camel case for consistency.